tests/e2e/support/cursor.rs

```
@@ -43,8 +52,29 @@ pub fn cursors(text: &str) -> (String, Cursors) {
         }
     }
 
+    assert_eq!(selections.len() % 2, 0);
+
+    assert!(matches!(selections.first(), Some(SelectionStackElement::Open(_)) | None));
+
+    selections.into_iter().tuple_windows().for_each(|window| match window {
+        (SelectionStackElement::Open(_), SelectionStackElement::Open(_))
+        | (SelectionStackElement::Close(_), SelectionStackElement::Close(_)) => {
+            panic!("selections should not overlap")
+        }
+        (SelectionStackElement::Open(start), SelectionStackElement::Close(end)) => {
+            cursors.add_selection(Range { start, end });
+        }
+        _ => {}
+    });
```

you're silently ignoring this input

```
<sel></sel></sel><sel>
```